game/scripts/vscripts/components/heroselection/heroselection.lua

```diff
@@ -935,12 +974,34 @@ function HeroSelection:GetPreviewHero (playerId)
   return
 end
 
+function HeroSelection:RandomHeroByAbility (ability, heroExclusions)
+  local attempts = 0
+  while true do
+    attempts = attempts + 1
+    local choice = HeroSelection:RandomHero()
+    if not heroExclusions[choice] and herolist[choice] == ability then
+      return choice
+    end
+
+    if attempts > 1000 then
```

Comment:
1000 attempts? Lua lags even after 10 attempts and you have 1000 lmao. Maybe put max amount of heroes (124+) or number of available non-banned heroes.

Reply:
this is a really right loop, it was infinite before to be fair haha